core/src/main/java/org/p2p/core/network/gson/GsonTypeAdapters.kt

```diff
@@ -213,3 +217,43 @@ internal class OptionalTypeAdapter<T : Any>(
         return Optional.of(gson.fromJson(reader, type))
     }
 }
+
+internal class Base64StringTypeAdapter() : TypeAdapter<Base64String?>() {
+    override fun write(writer: JsonWriter, value: Base64String?) {
+        if (value == null) {
+            writer.nullValue()
+        } else {
+            val stringValue = value.base64Value
+            writer.value(stringValue)
+        }
+    }
+
+    override fun read(reader: JsonReader): Base64String? {
+        if (reader.peek() == JsonToken.NULL) {
+            reader.nextNull()
+            return null
+        }
+        val stringValue = reader.nextString()
+        return stringValue?.toBase64Instance()
+    }
+}
+
+internal class Base58StringTypeAdapter() : TypeAdapter<Base58String?>() {
```

Comment:
have you checked org.p2p.core.network.gson.Base64TypeAdapter?

Reply:
oh great! couldn't find it